app/scripts/proactive/view/dom.js

```diff
@@ -326,6 +326,170 @@ define(
             });
         }
 
+        $("#menu-calling").click(function(event) {
+            //make the calling dropdown draggable
+            $("#span-calling-draggable").draggable({
+                helper: "original",
+                distance: 20,
+                stop: 200
+            });
+            //make the calling dropdown empty
+            $("#ul-calling").empty();
+            //add pin Open as first item in the dropdown
+            var pinOpen = $('<li id="li-pin-calling" role="presentation" class="dropdown-header"><img src="images/icon-pin.png"> Pin open</li>');
+            $("#ul-calling").append(pinOpen);
+            var menuElement = $('#ul-calling').parent();
+            if (!menuElement.hasClass('dropdown')) {
+                $("#li-pin-calling").html('<img src="images/icon-unpin.png"> Unpin');
+            }
+            //pinUnpin function pin/unpin the menu
+            $("#li-pin-calling").click(function(event) {
+                pinUnpin($("#ul-calling"), $("#li-pin-calling"));
+            });
+
+            event.preventDefault();
+            var studioApp = require('StudioApp');
+            var tasks = studioApp.models.jobModel.tasks;
+            var TasksArray = [];
+            for (var i = 0; i < tasks.length; i++) {
+                TasksArray[i] = tasks[i];
+            }
+            //isCallingObjects becomes true when there is at least one called catalog object
+            var isCallingObjects = false;
+            for (var i = 0; i < TasksArray.length; i++) {
+                var task = TasksArray[i];
+                if (task.has('Variables')) {
+                    var variables = task.get('Variables');
+                    var taskName = "previousTask";
+                    for (var j = 0; j < variables.length; j++) {
+                        var variable = variables[j];
+                        if (variable.Model && variable.Model.includes("PA:CATALOG_OBJECT")) {
+                            isCallingObjects = true;
+                            var tName = task.get('Task Name');
+                            //we add the task name only the first time, else we print empty string
+                            if (tName != taskName && taskName != "") {
+                                taskName = tName;
+                            } else {
+                                taskName = "";
+                            }
+                            var variableValue = variable.Value;
+                            if (variableValue) {
+                                var taskViewModel = studioApp.views.workflowView.taskViews[0];
+                                var calledObjectDetails = taskViewModel.getCalledObjectDetails(variableValue)
+                                var objectKind = taskViewModel.getObjectKind(calledObjectDetails["bucketName"], calledObjectDetails["objectName"])
+                                if (objectKind == 'Workflow/standard' || objectKind == 'Workflow/psa') {
```

Comment:
objectKind.indexOf('Workflow') == 0

you can do this also for similar tests in TaskView.js